### PR TITLE
Add sidekiq 7.2 to test matrix

### PR DESCRIPTION
**Pull request by rwojsznis**
Initial discovery on #21

Root cause: https://github.com/sidekiq/sidekiq/blob/main/Changes.md#720

> Adjust redis-client adapter to avoid method_missing [#6083] This can result in app code breaking if your app's Redis API usage was depending on Sidekiq's adapter to correct invalid redis-client API usage

7.2 changed their internal API regarding calling `redis-client`; Support to 7.0 was added by _community_ in #18 and I have very vague memory of it

### CODE REVIEW

**Comment by rwojsznis** [comment]
In case anyone is waiting for this - not forgotten, just got busy with everyday work; will try to keep this gem alive :)

**Comment by 9mm** [comment]
Awesome, thanks

**Comment by 9mm** [comment]
I def use this gem on like every project btw, so you know your work is awesome. It solves a really important need that for some reason isnt just built into sidekiq.

File: test/lib/lock_test.rb [hunk 1]
```diff
@@ -7,23 +7,24 @@ module Sidekiq
...
       cmd = 'sidekiq -r ./test/test_workers.rb -v'
-      buffer = ''
+      buffer_out = ''
 
       # very not fancy (https://78.media.tumblr.com/tumblr_lzkpw7DAl21qhy6c9o2_400.gif)
       # solution, but should do the job
       Open3.popen3(cmd) do |stdin, stdout, stderr, thread|
         begin
-          Timeout.timeout(5) do
+          Timeout.timeout(3) do
```
Comment (rwojsznis) [inline test/lib/lock_test.rb:16]: still no better idea for this one, leaving as it

File: test/test_workers.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+$LOAD_PATH.unshift(File.expand_path('../lib', __dir__))
```
Comment (rwojsznis) [inline test/test_workers.rb:1]: not sure how this worked before; seems like something changed with load paths I'm not aware of? require stopped working 🤔 will not investigate if the build is green tho 🙈

File: sidekiq-lock.gemspec [hunk 3]
```diff
@@ -18,10 +18,4 @@ Gem::Specification.new do |spec|
 
   spec.add_dependency "sidekiq", ">= 2.14.0"
   spec.add_dependency "redis",   ">= 3.0.5"
-
-  spec.add_development_dependency "bundler"
-  spec.add_development_dependency "rake"
-  spec.add_development_dependency "mocha", "~> 0.14.0"
-  spec.add_development_dependency "minitest"
-  spec.add_development_dependency "appraisal"
```
Comment (rwojsznis) [inline sidekiq-lock.gemspec:26]: unsure what's the best practice nowadays, seems like sidekiq itself defines all this stuff in Gemfile now

**Comment by rwojsznis** [comment]
All tests green; will release new version this week; dropped sidekiq 5 from the matrix [as it seems EOLed?](https://github.com/sidekiq/sidekiq/wiki/Commercial-Support#version-policy)

**Pull request merged**